### PR TITLE
feat | Added configuration options for "force create user" inside MagicLinkContinuation flow

### PR DESCRIPTION
**Pull request by alainkaiser**
- Boolean property to configure "force create user" also for MagicLinkContinuation flow
- Implementation done according to MagicLink flow
- Extracted common "is" method into AuthenticatorSharedUtils (used by MagicLinkContinuationAuthenticator & MagicLinkAuthenticator)

### CODE REVIEW

**Comment by alainkaiser** [comment]
Tried to install the .jar locally, works, but the new option is not displayed. I think some things are still missing.

**Comment by alainkaiser** [comment]
Sorry, my bad.
Inside the method `getConfigProperties` I had a typo (called `setType` twice instead of `setType` and `setName`.
Was able to test it locally. Works like expected:

image

I hope we can merge that :)

File: src/main/java/io/phasetwo/keycloak/magic/auth/MagicLinkContinuationAuthenticatorFactory.java [hunk 1]
```diff
@@ -42,32 +46,47 @@ public boolean isConfigurable() {
...
+            "Magic link authenticator expiration time in minutes. Default expiration period 10 minutes.");
     timeout.setDefaultValue("10");
 
-    return List.of(timeout);
-  }
-
-  @Override
-  public Authenticator create(KeycloakSession session) {
-    return new MagicLinkContinuationAuthenticator();
+    return Arrays.asList(createUser, timeout);
```
Comment (xgp) [inline src/main/java/io/phasetwo/keycloak/magic/auth/MagicLinkContinuationAuthenticatorFactory.java:89]: Prefer to keep this `List.of`

File: src/main/java/io/phasetwo/keycloak/magic/auth/util/AuthenticatorSharedUtils.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package io.phasetwo.keycloak.magic.auth.util;
+
+import org.keycloak.authentication.AuthenticationFlowContext;
+import org.keycloak.models.AuthenticatorConfigModel;
+
+import java.util.Map;
+
+public final class AuthenticatorSharedUtils {
```
Comment (xgp) [inline src/main/java/io/phasetwo/keycloak/magic/auth/util/AuthenticatorSharedUtils.java:8]: Nit. Prefer not to use `...Utils` for naming. Please call this class `Authenticators` per the guava standard.

File: src/main/java/io/phasetwo/keycloak/magic/auth/util/AuthenticatorSharedUtils.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    public static boolean is(AuthenticationFlowContext context, String propName, boolean defaultValue) {
+        AuthenticatorConfigModel authenticatorConfig = context.getAuthenticatorConfig();
+        if (authenticatorConfig == null) return defaultValue;
+
+        Map<String, String> config = authenticatorConfig.getConfig();
+        if (config == null) return defaultValue;
+
+        String v = config.get(propName);
+        if (v == null || "".equals(v)) return defaultValue;
```
Comment (xgp) [inline src/main/java/io/phasetwo/keycloak/magic/auth/util/AuthenticatorSharedUtils.java:17]: Prefer the guava method https://guava.dev/releases/snapshot-jre/api/docs/com/google/common/base/Strings.html#isNullOrEmpty(java.lang.String)

File: src/main/java/io/phasetwo/keycloak/magic/auth/util/AuthenticatorSharedUtils.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+        return v.trim().toLowerCase().equals("true");
```
Comment (xgp) [inline src/main/java/io/phasetwo/keycloak/magic/auth/util/AuthenticatorSharedUtils.java:19]: Prefer `return Boolean.parseBoolean(v.trim());`

File: src/main/java/io/phasetwo/keycloak/magic/auth/MagicLinkContinuationAuthenticator.java [hunk 5]
```diff
@@ -8,6 +8,7 @@
 
 import io.phasetwo.keycloak.magic.MagicLink;
 import io.phasetwo.keycloak.magic.auth.token.MagicLinkContinuationActionToken;
+import io.phasetwo.keycloak.magic.auth.util.AuthenticatorSharedUtils;
```
Comment (xgp) [inline src/main/java/io/phasetwo/keycloak/magic/auth/MagicLinkContinuationAuthenticator.java:11]: should be a static import. E.g. `import static io.phasetwo.keycloak.magic.auth.util.AuthenticatorSharedUtils.is;`

File: src/main/java/io/phasetwo/keycloak/magic/auth/MagicLinkContinuationAuthenticator.java [hunk 6]
```diff
@@ -31,6 +32,8 @@
 @JBossLog
 public class MagicLinkContinuationAuthenticator extends UsernamePasswordForm {
 
+  static final String CREATE_NONEXISTENT_USER_CONFIG_PROPERTY = "ext-magic-create-continuation-nonexistent-user";
```
Comment (xgp) [inline src/main/java/io/phasetwo/keycloak/magic/auth/MagicLinkContinuationAuthenticator.java:35]: Probably a good time to make this consistent. Move the prop keys to the `MagicLink` class and reuse the naming in both places.

**Review by xgp — CHANGES_REQUESTED** [review]
Thank you for the PR.

Couple of requested changes.

Also, can you please run `mvn fmt:format` before committing?

File: src/main/java/io/phasetwo/keycloak/magic/auth/MagicLinkContinuationAuthenticatorFactory.java (hunk 1 above)
Comment (alainkaiser) [reply src/main/java/io/phasetwo/keycloak/magic/auth/MagicLinkContinuationAuthenticatorFactory.java]: Changed to `List.of` in both Factories.

File: src/main/java/io/phasetwo/keycloak/magic/auth/util/AuthenticatorSharedUtils.java (hunk 2 above)
Comment (alainkaiser) [reply src/main/java/io/phasetwo/keycloak/magic/auth/util/AuthenticatorSharedUtils.java]: Changed naming.

File: src/main/java/io/phasetwo/keycloak/magic/auth/util/AuthenticatorSharedUtils.java (hunk 3 above)
Comment (alainkaiser) [reply src/main/java/io/phasetwo/keycloak/magic/auth/util/AuthenticatorSharedUtils.java]: Did not know them, very nice. Changed to it.

File: src/main/java/io/phasetwo/keycloak/magic/auth/util/AuthenticatorSharedUtils.java (hunk 4 above)
Comment (alainkaiser) [reply src/main/java/io/phasetwo/keycloak/magic/auth/util/AuthenticatorSharedUtils.java]: done.

File: src/main/java/io/phasetwo/keycloak/magic/auth/MagicLinkContinuationAuthenticator.java (hunk 5 above)
Comment (alainkaiser) [reply src/main/java/io/phasetwo/keycloak/magic/auth/MagicLinkContinuationAuthenticator.java]: Added static imports for both of the Authenticator to keep it consistent.

File: src/main/java/io/phasetwo/keycloak/magic/auth/MagicLinkContinuationAuthenticator.java (hunk 6 above)
Comment (alainkaiser) [reply src/main/java/io/phasetwo/keycloak/magic/auth/MagicLinkContinuationAuthenticator.java]: Added it to the `MagicLink` class, replaced all usages.

**Comment by alainkaiser** [comment]
@xgp Thanks for the review. Your comments should be resolved. Also run formatting before the commit as request.

**Review by xgp — APPROVED** [review]
Thanks for making the changes. LGTM!

**Pull request merged**